src/legacy/core_plugins/timelion/public/shim/legacy_dependencies_plugin.ts

```diff
@@ -0,0 +1,56 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import chrome from 'ui/chrome';
+import { Plugin } from 'kibana/public';
+// @ts-ignore
+import { visFactory } from 'ui/vis/vis_factory';
+import { initTimelionLegacyModule } from './timelion_legacy_module';
+
+/** @internal */
+export interface LegacyDependenciesPluginSetup {
```

Comment:
visFactory shouldn't be a legacy dependency, its statically exported from `visualizations` plugin already, so

`import { visFactory } from `../../visualizations/public`, and you can import it directly where you need it (no need to pass it down) as its stateless function.

Reply:
Moved